streams/test-utils/src/test/java/org/apache/kafka/streams/TopologyTestDriverTest.java

```diff
@@ -897,4 +898,70 @@ public void close() {}
             );
         }
     }
+
+    @Test
+    public void shouldProcessFromSourcesThatMatchMultiplePattern() {
+        final  String patternSourceTopic3 = "source-topic-[A-Z]";
+        final  String consumerTopic3 = "source-topic-Z";
+
+        final ConsumerRecord<byte[], byte[]> consumerRecord3 = consumerRecordFactory.create(consumerTopic3, key1, value1, timestamp1);
+
+        testDriver = new TopologyTestDriver(setupMultipleSourceTopology(PATTERN_SOURCE_TOPIC_1, SOURCE_TOPIC_2, patternSourceTopic3), config);
+
+        final List<Record> processedRecords1 = mockProcessors.get(0).processedRecords;
+        final List<Record> processedRecords2 = mockProcessors.get(1).processedRecords;
+        final List<Record> processedRecords3 = mockProcessors.get(2).processedRecords;
+
+        testDriver.pipeInput(consumerRecord1);
+
+        assertEquals(1, processedRecords1.size());
+        assertEquals(0, processedRecords2.size());
+        assertEquals(0, processedRecords3.size());
+
+        Record record = processedRecords1.get(0);
+        Record expectedResult = new Record(consumerRecord1);
+        expectedResult.offset = 0L;
+        assertThat(record, equalTo(expectedResult));
+
+        testDriver.pipeInput(consumerRecord2);
+
+        assertEquals(1, processedRecords1.size());
+        assertEquals(1, processedRecords2.size());
+        assertEquals(0, processedRecords3.size());
+
+        record = processedRecords2.get(0);
+        expectedResult = new Record(consumerRecord2);
+        expectedResult.offset = 0L;
+        assertThat(record, equalTo(expectedResult));
+
+        testDriver.pipeInput(consumerRecord3);
+
+        assertEquals(1, processedRecords1.size());
+        assertEquals(1, processedRecords2.size());
+        assertEquals(1, processedRecords3.size());
+
+        record = processedRecords3.get(0);
+        expectedResult = new Record(consumerRecord3);
+        expectedResult.offset = 0L;
+        assertThat(record, equalTo(expectedResult));
+
+    }
+
+    @Test
+    public void shouldProcessFromSourceThatMatchPattern() {
+        final String sourceName = "source";
+
+        final Topology topology = new Topology();
+
+        topology.addSource(sourceName, PATTERN_SOURCE_TOPIC_1);
```

Comment:
You need to pass in a `Pattern` but not a `String` here.

This actually exposes a "bug" in the test-driver -- it should check if the topic name is valid -- and a pattern is not a valid topic name. Not sure if we should have a different PR for a fix or piggy back on this PR. \cc @guozhangwang @bbejeck @vvcephei

Reply:
Let's piggy-back on this PR: it should not be a big fix.